### PR TITLE
ADR016: two-eyes process

### PR DESCRIPTION
**Pull request by chrisfarms**
## What

Add ADR for two-eyes process

### CODE REVIEW

File: adr/ADR016-two-eyes-git-verification.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# ADR015: Verifing two-eyes on code changes
```
Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: change to ADR016

File: adr/ADR016-two-eyes-git-verification.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+without being exposed to at least two authorized code owners.
+
+Our aim is to improve on process-based appoval methods by being able to
+digitally verify that code has been exposed to at least two authorized code
+owners without adversely affecting developer workflow.
+
+Some potential solutions involve:
+
+1. Enforcing that two unique developers each an add empty signed commit at the
+   tip of the branch for PRs before being merged, and verifying the existance
```
Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: typo existence

File: adr/ADR016-two-eyes-git-verification.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# ADR015: Verifing two-eyes on code changes
+
+## Status
+
+Draft
```
Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: Change to Accepted

File: adr/ADR016-two-eyes-git-verification.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Context
+
+All of our deployment processes start with code in git repositories hosted on
+Github and continous delivery pipelines automate the build and deployment of
+applications out to production. Git supports GPG signing of commits.
+
+Code changes should not be able to make it out to a production environment
+without being exposed to at least two authorized code owners.
```
Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: typo authorised

Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: typo authorised

File: adr/ADR016-two-eyes-git-verification.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+Our aim is to improve on process-based appoval methods by being able to
+digitally verify that code has been exposed to at least two authorized code
```
Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: typo authorised

File: adr/ADR016-two-eyes-git-verification.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+protection against developer machine compromise than Github's session duration
+alone.  However we don't feel this benefit warrants the reduced usability for
+developers across GDS.
+
+We implement solution (3).
+
+## Consequences
+
+* Developer workflow is largely unaffected across GDS by taking advantage of
+  Github features team are already familer
```
Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: typo familiar

Comment (pauldougan) [inline adr/ADR016-two-eyes-git-verification.md]: I have reviewed the notes in the trello card and the ADR is consistent with that.

Comment (paroxp) [reply adr/ADR016-two-eyes-git-verification.md]:
```suggestion
  Github features team are already familiar
```

File: adr/ADR016-two-eyes-git-verification.md (hunk 5 above)
Comment (paroxp) [reply adr/ADR016-two-eyes-git-verification.md]:
```suggestion
digitally verify that code has been exposed to at least two authorised code
```

File: adr/ADR016-two-eyes-git-verification.md (hunk 4 above)
Comment (paroxp) [reply adr/ADR016-two-eyes-git-verification.md]:
```suggestion
without being exposed to at least two authorised code owners.
```

Comment (paroxp) [reply adr/ADR016-two-eyes-git-verification.md]:
```suggestion
without being exposed to at least two authorised code owners.
```

File: adr/ADR016-two-eyes-git-verification.md (hunk 3 above)
Comment (paroxp) [reply adr/ADR016-two-eyes-git-verification.md]:
```suggestion
Accepted
```

File: adr/ADR016-two-eyes-git-verification.md (hunk 2 above)
Comment (paroxp) [reply adr/ADR016-two-eyes-git-verification.md]:
```suggestion
   tip of the branch for PRs before being merged, and verifying the existence
```

File: adr/ADR016-two-eyes-git-verification.md (hunk 1 above)
Comment (paroxp) [reply adr/ADR016-two-eyes-git-verification.md]:
```suggestion
# ADR016: Verifying two-eyes on code changes
```

**Pull request merged**